src/OrbitGl/MockBatcher.h

```diff
@@ -27,18 +27,25 @@ class MockBatcher : public Batcher {
   void ResetElements() override;
   [[nodiscard]] uint32_t GetNumElements() const override;
 
-  [[nodiscard]] std::vector<float> GetLayers() const override { return {}; };
+  [[nodiscard]] std::vector<float> GetLayers() const override {
+    return std::vector<float>(z_layers_.begin(), z_layers_.end());
+  }
+
   virtual void DrawLayer(float /*layer*/, bool /*picking*/) const override{};
 
   [[nodiscard]] const PickingUserData* GetUserData(PickingId /*id*/) const override {
     return nullptr;
   }
 
-  [[nodiscard]] int GetNumLinesByColor(Color color) const { return num_lines_by_color_.at(color); }
+  [[nodiscard]] int GetNumLinesByColor(Color color) const {
+    return num_lines_by_color_.count(color) ? num_lines_by_color_.at(color) : 0;
+  }
   [[nodiscard]] int GetNumTrianglesByColor(Color color) const {
-    return num_triangles_by_color_.at(color);
+    return num_triangles_by_color_.count(color) ? num_triangles_by_color_.at(color) : 0;
+  }
+  [[nodiscard]] int GetNumBoxesByColor(Color color) const {
+    return num_boxes_by_color_.count(color) ? num_boxes_by_color_.at(color) : 0;
```

Comment:
for the future: absl::flat_hash_map instead of std::map for 2 reasons. 
1. General performance go/totw/183
2. Here is an implicit conversion from int (or size_t) to bool. So this should either be `num_boxes_by_color_.count(color) == 0` or better with absl `num_boxes_by_color_.contains(color)`

Reply:
as discussed offline, we will use a BtreeMap